nomad/structs/structs.go

```diff
@@ -3943,6 +3943,8 @@ func (j *Job) Validate() error {
 		mErr.Errors = append(mErr.Errors, errors.New("Missing job ID"))
 	} else if strings.Contains(j.ID, " ") {
 		mErr.Errors = append(mErr.Errors, errors.New("Job ID contains a space"))
+	} else if strings.Contains(j.ID, "\000") {
+		mErr.Errors = append(mErr.Errors, errors.New("Job ID contains a null chararacter"))
 	}
 	if j.Name == "" {
```

Comment:
It doesn't come up except when using the API, but Jobs have both a name and an ID. Should we be validating that `Job.Name` also doesn't contain a null char?

Reply:
yeah, good point. i don't see any indexing there, but a quick test showed that null in `Name` put a docker job into a failing state:
```
2020-10-05T12:55:36Z  Driver Failure  Failed to start container cc77616db08d0701b20b438af789855e56eb343f7ce9c0636c779e399e052177: API error (500): OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"setenv: invalid argument\"": unknown
```

will update PR to include `Job.Name`

Reply:
ooh, in addition, the task drivers set the `NOMAD_DC` environment variable, which will fail if the node datacenter includes a null character 🤦

Reply:
and region 🤦++